runtime/runtime_storage.go

```diff
@@ -156,38 +171,50 @@ func (s *interpreterRuntimeStorage) writeValue(
 	key string,
 	value interpreter.OptionalValue,
 ) {
-	storageKey := storageKey{
+	fullKey := storageKey{
 		storageIdentifier: storageIdentifier,
 		key:               key,
 	}
 
 	// Only write the value to the cache.
 	// The Cache is finally written back through the runtime interface in `writeCached`
 
+	entry := s.cache[fullKey]
+	entry.mustWrite = true
```

Comment:
RE: naming, curious what is better in this case `entry.mustWrite` or something like `entry.modified`/`entry.dirty`. Super minor!